lib/jwt.js

```diff
@@ -0,0 +1,202 @@
+var urlDecodeB64 = function(data) {
+  return new Buffer(data, 'base64').toString('ascii');
+}
+
+var isNumber = function(n) { return typeof n === 'number' };
+
+var idTokendecoded = [
+  'iss',
+  'aud',
+  'exp',
+  'nbf',
+  'iat',
+  'jti',
+  'azp',
+  'nonce',
+  'auth_time',
+  'at_hash',
+  'c_hash',
+  'acr',
+  'amr',
+  'sub_jwk',
+  'cnf',
+  'sip_from_tag',
+  'sip_date',
+  'sip_callid',
+  'sip_cseq_num',
+  'sip_via_branch',
+  'orig',
+  'dest',
+  'mky',
+  'events',
+  'toe',
+  'txn',
+  'rph',
+  'sid',
+  'vot',
+  'vtm'
+];
+
+var decode = function(token) {
+  var parts = token.split('.');
+  var [header, payload, signature] = parts;
+
+  if (parts.length !== 3 || !header || !payload || !signature) {
+    throw new Error('ID token could not be decoded');
+  }
+  var payloadJSON = JSON.parse(urlDecodeB64(payload));
+  var claims = { __raw: token };
+  var user = {};
+  Object.keys(payloadJSON).forEach(function(k) {
+    claims[k] = payloadJSON[k];
+    if (!idTokendecoded.includes(k)) {
+      user[k] = payloadJSON[k];
+    }
+  });
+  return {
+    encoded: { header, payload, signature },
+    header: JSON.parse(urlDecodeB64(header)),
+    claims: claims,
+    user: user
+  };
+};
+
+var verify = function(idToken, options) {
+  if (!idToken) {
+    throw new Error('ID token is required but missing');
+  }
+
+  var decoded = decode(idToken);
```

Comment:
Unless I'm missing something ... I'm not seeing the ID token signature validated during this process.